### DISCUSSION TITLE
What happens when I've reached the GitHub Codespaces free plan limit?

### DISCUSSION THREAD

**Discussion by ghost** [community/community#39670] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hi everyone. 

I'm thinking of using GitHub codespaces with the free plan. I would like to know what happens when I reach the 15GB free plan. For example, is my account blocked? Another question, if I remove the files, will I have 15GB of space again?

**Comment by spenserblack** [comment, marked as answer]
I ended up maxing out my storage very quickly when I first started using Codespaces. My account was not blocked, but I was unable to access my existing codespaces to do any more work in them. However, GitHub gives you the option to copy changes that are in your codespace to a branch, so you don't have to worry too much about losing work.

>if I remove the files, will I have 15GB of space again?

No, it doesn't reset until the next billing cycle. It's not "how much storage are you using", but "how much storage did you use, and for how long?" [This comment provides the formula](https://github.com/community/community/discussions/39697#discussioncomment-4335544). AFAIK all codespaces on the free plan are 32GB.

**Reply by ghost** [reply]
Hi spenserblack.

> I ended up maxing out my storage very quickly when I first started using Codespaces. My account was not blocked, but I was unable to access my existing codespaces to do any more work in them. However, GitHub gives you the option to copy changes that are in your codespace to a branch, so you don't have to worry too much about losing work.
> No, it doesn't reset until the next billing cycle. It's not "how much storage are you using", but "how much storage did you use, and for how long?" https://github.com/community/community/discussions/39697#discussioncomment-4335544. AFAIK all codespaces on the free plan are 32GB.

thank you for feedback.
